share/wake/lib/system/runner.wake

```diff
@@ -55,97 +56,70 @@ export tuple RunnerOutput =
 
-    Runner name score doit
+    Runner name do
 
-# This runner does not detect inputs/outputs on it's own
-# You must use Fn{Inputs,Outputs} to fill in this information
+# localRunner: A runner that provides no sandboxing protections and no file tracking.
+#
+# You must use Fn{Inputs,Outputs} to fill in this information for wake to maintain safety and reusability
+# Advanced usage only, proceed with caution
 export def localRunner: Runner =
-    def launch job dir stdin env cmd status runtime cputime membytes ibytes obytes isatty =
-        prim "job_launch"
+    def run (job: Job) ((RunnerInput _ cmd vis env dir stdin _ _ predict isatty): RunnerInput): Result RunnerOutput Error =
+        def jobKey = JobKey dir stdin env.implode cmd.implode 0 "" isatty.booleanToInteger
+        def _ = primJobLaunch job jobKey predict
 
-    def badlaunch job error = prim "job_fail_launch"
+        job
+        | getJobReality
+        |< (\reality RunnerOutput (vis | map getPathName) Nil reality)
 
-    def doit job = match _
-        Fail e ->
-            def _ = badlaunch job e
-
-            Fail e
-        Pass (RunnerInput _ cmd vis env dir stdin _ _ predict isatty) ->
-            def Usage status runtime cputime mem in out = predict
-
-            def _ =
-                launch
-                job
-                dir
-                stdin
-                env.implode
-                cmd.implode
-                status
-                runtime
-                cputime
-                mem
-                in
-                out
-                (booleanToInteger isatty)
-
-            match (getJobReality job)
-                Pass reality -> Pass (RunnerOutput (map getPathName vis) Nil reality)
-                Fail f -> Fail f
-
-    def score _ = Pass 1.0
-
-    Runner "local" score doit
+    makeRunner "local" run
 
+# virtualRunner: A runner that immediatly marks the job as complete using the predicted usage
+#
+# This runner is useful for tracking a unit of work that is job like but not launched as a process
 export def virtualRunner: Runner =
-    def virtual job stdout stderr status runtime cputime membytes ibytes obytes = prim "job_virtual"
-    def badlaunch job error = prim "job_fail_launch"
+    def run (job: Job) ((RunnerInput _ _ vis _ _ _ _ _ predict _): RunnerInput): Result RunnerOutput Error =
+        def _ = primJobVirtual job "" "" predict
 
-    def doit job = match _
-        Fail e ->
-            def _ = badlaunch job e
+        job
+        | getJobReality
+        |< (\reality RunnerOutput (vis | map getPathName) Nil reality)
 
-            Fail e
-        Pass (RunnerInput _ _ vis _ _ _ _ _ predict _) ->
-            def Usage status runtime cputime mem in out = predict
+    makeRunner "virtual" run
 
-            def _ =
-                virtual job "" "" status runtime cputime mem in out # sets predict+reality
+# depreMakeRunner: Deprecated. Do not use
+export def depreMakeRunner name pre post (Runner _ run) =
```

Comment:
There's definitely arguments either way for making the deprecation obvious in the name, but if we're saying we want to deprecate this rather than deleting it immediately, I'd still prefer having a bit more descriptive a name of what it's actually doing.  The comment and conversation should be sufficient to let people know not to use it.
```suggestion
export def wrapRunner name pre post (Runner _ run) =
```

Reply:
Ah sure. FWIW it probably only keeping this for one release. I've already written the internal changes to migrate off of this. Its more as a warning for anyone external who depends on it